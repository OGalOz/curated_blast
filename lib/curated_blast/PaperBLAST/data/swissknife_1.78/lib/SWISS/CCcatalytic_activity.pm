package SWISS::CCcatalytic_activity;

use vars qw($AUTOLOAD @ISA %fields);

use Carp;
use strict;
use SWISS::TextFunc;
use SWISS::BaseClass;
use SWISS::CC;


BEGIN {
    @ISA = ( 'SWISS::BaseClass' );

    %fields = (
        activity   => undef, # {  rhea => "", ec => "", reaction => "", participants => [], ev => "" }
        directions => undef, # [ { txt => "", rhea => "", ev => "" } ]
        form       => undef,
        _old       => undef  # [ [ txt, ev ], ]
    );
}


sub new {
    my $ref   = shift;
    my $class = ref($ref) || $ref;
    my $self  = new SWISS::BaseClass;
    $self->rebless($class);
    return $self;
}


sub fromText {
    my $class   = shift;
    my $textRef = shift;
    my $self    = new SWISS::CCcatalytic_activity;
    my $text    = $$textRef;
    $self->initialize();
    $text =~ s/ *-!- CATALYTIC ACTIVITY: +//;
    $text =~ s/ {2,}/ /g;

    if ( $text =~ /Reaction=/ ) { # new format (rhea/ec & reaction)
        my $core = {}; my $iscore = 0;
        foreach my $elem ( split /; /, $text ) {
            $elem =~ s/;$//;
            my ( $key, $val ) = ( "", "" );
            ( $key, $val ) = $elem =~ /^(\w+)=(.+)$/;
            if ( $key eq "Reaction" ) {
                $iscore = 1;
                $core->{reaction} = $val;
            }
            elsif ( $key eq "Xref" ) {
                my @ids = split /, /, $val;
                ( my $rhea_id = shift @ids ) =~ s/^Rhea://;
                if ( $iscore ) { # core (reaction)
                    $core->{rhea} = $rhea_id;
                    $core->{participants} = \@ids;
                }
                else           { # direction
                    my $dir = $self->{ directions }->[ -1 ] || { 'txt' => undef, 'ev' => undef };
                    $dir->{rhea} = $rhea_id;
                }
            }
            elsif ( $key eq "EC" ) {
                $core->{ec} = $val;
            }
            elsif ( $key eq "Evidence" ) {
                if ( $iscore ) { $core->{ev} = $val } # core (reaction)
                else           { # direction
                    my $dir = $self->{ directions }->[ -1 ] || { 'txt' => undef, 'rhea' => undef };
                    $dir->{ev} = $val;
                }
            }
            elsif ( $key eq "PhysiologicalDirection" ) {
                $iscore = 0;
                push @{ $self->{directions} }, { 'txt' => $val, 'rhea' => undef, 'ev' => undef };
            }
        }
        $self->{activity} = $core;
    }
    else { # old unstructured format
        my $has_new_style_ev = $text =~ /ECO:/;
        $self->{_old} = SWISS::CC::parse2Blocks( $text );
        # set evidenceTags 4 compatibility with existing ...EvidenceTag methods in BaseClass! + old tests (fixme: clean/remove that mess!?)
        # but the real evidences are now within blocks (2nd field of block array)
        my @evs = map { split /, ?/, $_->[1] } grep { $_->[1] } @{ $self->{_old} };
        my $ev  = $has_new_style_ev ? " {" . join( ", ", @evs ) . "}" : "{" . join( ",", @evs ) . "}";
        $self->evidenceTags( $ev ) if @evs;
    }

    $self->{ _dirty } = 0;
    return $self;
}


sub topic {
    return "CATALYTIC ACTIVITY";
}


sub activity_rhea { # get (or set) rhea (id without "Rhea:" prefix, but with RHEA:) e.g. (RHEA:18037)
    my ( $self, $value, $ev ) = @_;

    if ( !$self->{_old} ) { # only works with new (rhea/ec + reaction) format
        if ( defined $value ) {
            my $new_ev = $ev ? $ev : $self->{ activity }->{ev};
            $self->{activity}->{rhea} = $value;
            $self->{activity}->{ev}   = $new_ev;
        }

        return $self->{activity} ? $self->{activity}->{rhea} : "";
    }
    else { return "" }
}

sub activity_reaction { # get (or set) reaction
    my ( $self, $value, $ev_only_for_old_format ) = @_;

    if ( !$self->{_old} ) { # new (rhea/ec + reaction) format
        if ( defined $value ) {
            $self->{activity}->{reaction} = $value;
        }

        return $self->{activity} ? $self->{activity}->{reaction} : "";
    }
    else { # old format
        if ( defined $value ) {
            my $new_ev = $ev_only_for_old_format ? $ev_only_for_old_format : " {" . join( ", ", map { split /, ?/, $_->[1] } grep { $_->[1] } @{ $self->{_old} } ) . "}";
            $self->evidenceTags( $new_ev );
            $self->{_old} = [ [ $value, $new_ev ] ];
        } # -> ! not tested
        return join( ". ", map { $_->[0] } @{ $self->{_old} } )."."
    }
}

sub activity_ec { # get (or set) ec
    my ( $self, $value, $ev ) = @_;

    if ( !$self->{_old} ) { # only works with new (rhea/ec + reaction) format
        if ( defined $value ) {
            my $new_ev = $ev ? $ev : $self->{ activity }->{ev};
            $self->{activity}->{ec} = $value;
            $self->{activity}->{ev} = $new_ev;
        }

        return $self->{activity} ? $self->{activity}->{ec} : "";
    }
    else { return "" }
}

sub activity_participant {
    my ( $self, $values ) = @_;

    if ( !$self->{_old} ) { # only works with new (rhea/ec + reaction) format
        if (defined $values) {
            $self->{activity}->{participants} = $values;
        }
        return $self->{activity} ? $self->{activity}->{participants} : [];
    }
    else { return [] }
}

sub directions { # get (or set) direction(s) structures
    my ( $self, $values ) = @_;

    if ( !$self->{_old} ) { # only works with new (rhea/ec + reaction) format
        if ( defined $values && ref $values eq "ARRAY") {
            $self->{directions} = $values;
        }
        return $self->{directions};
    }
    else { return [] }
}


sub toString {
    my ( $self ) = @_;

    my $activity_txt = $self->_activity_to_text();

    my $text = "CC   -!- CATALYTIC ACTIVITY:";
    if ( !$self->{_old} ) { # new (rhea/ec, reaction) format
        #my $form = $self->{form} ? " $self->{form}:" : "";
        $text .= "\n";
        $text .= SWISS::TextFunc->wrapOn( 'CC       ',"CC         ", $SWISS::TextFunc::lineLength, $activity_txt );
        map {
            my $direction_txt = _direction_struct_to_text( $_ );
            $text .= SWISS::TextFunc->wrapOn( 'CC       ',"CC         ", $SWISS::TextFunc::lineLength, $direction_txt );
        } @{ $self->{directions} };
    }
    else { # old unstructured format
        $text .= " ".$activity_txt;
        $text = SWISS::TextFunc->wrapOn('',"CC       ", $SWISS::TextFunc::lineLength, $text );
    }

    return $text;
}

sub _activity_to_text {
    my ( $self, $no_ev ) = @_;
    if ( !$self->{_old} ) { # new structured format
        my $activity     = $self->{ activity };
        my $activity_txt = "Reaction=$activity->{reaction};";
        if ( $activity->{rhea} ) {
            $activity_txt .= " Xref=Rhea:$activity->{rhea}";
            if ( $activity->{participants} )  { $activity_txt .= ", " . join( ", ", @{ $activity->{participants} } ) }
            $activity_txt .= ";";
        }
        if ( $activity->{ec} )            { $activity_txt .= " EC=$activity->{ec};" }
        if ( $activity->{ev} && !$no_ev ) { $activity_txt .= " Evidence=$activity->{ev};" }
        return $activity_txt;
    }
    else { # old format
        return ( $no_ev ? join( ". ", map { $_->[0] } @{ $self->{_old} } )."." : SWISS::CC::blocks2String( $self->{_old}, $self->evidenceTags )."." );
    }
}
sub _direction_struct_to_text {
    my ( $direction, $no_ev ) = @_;
    my $direction_txt = "";
    if ( $direction->{txt} )           { $direction_txt .= "PhysiologicalDirection=$direction->{txt};" }
    if ( $direction->{rhea} )          { $direction_txt .= " Xref=Rhea:$direction->{rhea};" }
    if ( $direction->{ev} && !$no_ev ) { $direction_txt .= " Evidence=$direction->{ev};" }
    return $direction_txt;
}

sub comment { # unwrapped content (without ev)
    my ($self, $with_ev) = @_;

    my $activity_txt  = $self->_activity_to_text (  "no_ev=true" );
    my $direction_txt = "";
    map { $direction_txt .= ( $direction_txt ? " " : "" ) . _direction_struct_to_text( $_ , "no_ev=true" ); } @{ $self->{directions} };
    return $activity_txt . ( $direction_txt ? " $direction_txt" : "" );
}


#sub sort {
#    my $self = shift;
#    $self->{ list } = sort { lc( $a->[0] ) cmp lc( $b->[0] ) } @{ $self->{ list } };
#}


1;

__END__

=head1 Name

SWISS::CCcatalytic_activity

=head1 Description

B<SWISS::CCcofactor> represents a comment on the topic 'CATALYTIC ACTIVITY'
within a Swiss-Prot or TrEMBL entry as specified in the user manual
http://www.expasy.org/sprot/userman.html .  Comments on other topics are stored
in other types of objects, such as SWISS::CC (see SWISS::CCs for more information).

Collectively, comments of all types are stored within a SWISS::CCs container
object.

=head1 Inherits from

SWISS::BaseClass

=head1 Attributes

=over

=item topic

The topic of this comment ('CATALYTIC ACTIVITY').

=item comment

The "text" version of this comment (without evidences and new lines).


=back
=head1 Methods

=item activity_rhea

The rhea id (e.g. RHEA:18037) of the reaction (only for new structured CC CATALYTIC ACTIVITY)

=item activity_rhea( $new_rhea_id, $new_optional_reaction_ev )

Set rhea activity to $new_rhea_id (and reaction ev to $new_optional_reaction_ev if provided)
(only for new structured CC CATALYTIC ACTIVITY)

=item activity_reaction

The reaction txt (e.g. agmatine + H2O = N-carbamoylputrescine + NH4(+))

=item activity_reaction( $new_reaction_txt )

Set reaction txt to $new_reaction_txt
(on old unstructured format: extra $ev could be added)

=item activity_ec

The reaction EC (e.g. 3.5.3.12)

=item activity_ec( $new_reaction_ec, , $new_optional_reaction_ev )

Set reaction EC to $new_reaction_ec (and reaction ev to $new_optional_reaction_ev if provided)
(only for new structured CC CATALYTIC ACTIVITY)

=item activity_participant

The reaction participants as a String Array (e.g. [ "ChEBI:CHEBI:15377", "ChEBI:CHEBI:28938", "ChEBI:CHEBI:58145", "ChEBI:CHEBI:58318" ])
(participant are shown as dbtype:id e.g. in "ChEBI:CHEBI:15377" CheBI is db type, CHEBI:15377 is (CheBi) id)
(only for new structured CC CATALYTIC ACTIVITY)

=item activity_participant( $new_participant_array )

Set reaction participants to $new_participant_array (array of strings)
(only for new structured CC CATALYTIC ACTIVITY)


=item directions

The directions: Array of direction structures as hash e.g. [ { 'txt' => 'left-to-right', 'rhea' => 'RHEA:16846', 'ev' => '{ECO:0000269|PubMed:29420286}' } ]
p.s. in well formed UniProtKB data 'txt' will be 'right-to-left' or 'left-to-right' (in that order); there should be no more than 2 directions (minimum: 0)

=item direction( $new_directions )

Set directions Array to $new_directions ...
(only for new structured CC CATALYTIC ACTIVITY)

=head2 Standard methods

=over

=item new

=item fromText

=item toString

Returns a string representation of this comment.

=back
