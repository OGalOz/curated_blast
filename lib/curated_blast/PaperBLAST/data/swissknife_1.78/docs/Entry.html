<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>SWISS::Entry</title>
<meta http-equiv="content-type" content="text/html; charset=utf-8" />
<link rev="made" href="mailto:operations@isb-sib.ch" />
</head>

<body style="background-color: white">


<!-- INDEX BEGIN -->
<div name="index">
<p><a name="__index__"></a></p>

<ul>

	<li><a href="#name">Name</a></li>
	<li><a href="#description">Description</a></li>
	<li><a href="#example">Example</a></li>
	<li><a href="#attributes">Attributes</a></li>
	<li><a href="#methods">Methods</a></li>
	<ul>

		<li><a href="#data_access_methods">Data access methods</a></li>
	</ul>

</ul>

<hr name="index" />
</div>
<!-- INDEX END -->

<p>
</p>
<h1><a name="name">Name</a></h1>
<p>SWISS::Entry</p>
<p>
</p>
<hr />
<h1><a name="description">Description</a></h1>
<p>Main module to handle SWISS-PROT entries. One Entry object represents one SWISS-PROT entry and provides an API for its modification.</p>
<p>The basic concept is the idea of lazy parsing. If an Entry object is created from the entry in flat file format, the text is simply stored in the private text attribute of the entry object. The member objects of the entry are only created if they are dereferenced.</p>
<p>
</p>
<hr />
<h1><a name="example">Example</a></h1>
<PRE>
use SWISS::Entry;
# Read an entire record at a time
$/ = "\/\/\n";
while (<>){
  $entry = SWISS::Entry->fromText($_);
  print $entry->AC, "\n";
}
</pre><p>This minimum program reads entries from a file in SWISS-PROT format and prints the primary accession number for each of the entries.</p>
<p>
</p>
<hr />
<h1><a name="attributes">Attributes</a></h1>
<p>The following attributes represent member objects. They can be accessed like e.g. $entry-&gt;IDs</p>
<dl>
<dt><strong><a name="ids" class="item">IDs</a></strong></dt>

<dd>
<p>ID line object</p>
</dd>
<dt><strong><a name="acs" class="item">ACs</a></strong></dt>

<dt><strong><a name="dts" class="item">DTs</a></strong></dt>

<dt><strong><a name="des" class="item">DEs</a></strong></dt>

<dt><strong><a name="gns" class="item">GNs</a></strong></dt>

<dt><strong><a name="oss" class="item">OSs</a></strong></dt>

<dt><strong><a name="ocs" class="item">OCs</a></strong></dt>

<dt><strong><a name="refs" class="item">Refs</a></strong></dt>

<dd>
<p>The reference block object</p>
</dd>
<dt><strong><a name="ccs" class="item">CCs</a></strong></dt>

<dt><strong><a name="kws" class="item">KWs</a></strong></dt>

<dt><strong><a name="drs" class="item">DRs</a></strong></dt>

<dt><strong><a name="fts" class="item">FTs</a></strong></dt>

<dt><strong><a name="stars" class="item">Stars</a></strong></dt>

<dd>
<p>Object for the annotator's section stored in the ** lines.</p>
</dd>
<dt><strong><a name="sqs" class="item">SQs</a></strong></dt>

<dd>
<p>The sequence object.</p>
</dd>
</dl>
<p>
</p>
<hr />
<h1><a name="methods">Methods</a></h1>
<dl>
<dt><strong><a name="new" class="item">new</a></strong></dt>

<dd>
<p>Return a new Entry object</p>
</dd>
<dt><strong><a name="initialize" class="item">initialize</a></strong></dt>

<dd>
<p>Initialise an Entry object and return it.</p>
</dd>
<dt><strong><a name="update_force" class="item">update [force]</a></strong></dt>

<dd>
<p>Update an
entry. The content of the member objects is written back into the private text
attribute of the entry if necessary. If $force is true, an update of all 
member objects is forced.</p>
</dd>
<dt><strong><a name="reformat" class="item">reformat</a></strong></dt>

<dd>
<p>Reformat all fields of an entry.</p>
</dd>
<dt><strong><a name="fromtext_text_fullparse_removeinternalcomments" class="item">fromText $text [, $fullParse[, $removeInternalComments]]</a></strong></dt>

<dd>
<p>Create an Entry object from the text $text. If $fullParse is true, the entry is
parsed at creation time. Otherwise the individual line objects are only 
created if they are dereferenced. If $removeInternalComments is true, wild comments
and indentation will be removed from the text before the parsing is done. [NOTE: 
wild comments are lines starting with a double asterisk located outside the Stars
section, and indented lines are lines starting with spaces. Both are used internally
by SWISS-PROT annotators during their work and excluded from internal and external
releases.]</p>
</dd>
<dt><strong><a name="totext_insertinternalcomments" class="item">toText [$insertInternalComments]</a></strong></dt>

<dd>
<p>Return the entry in flat file text format. If internal comments and indentation
have been removed as specified in the parameters to <code>fromText()</code>, you may wish
to reinsert them in the text output by setting $insertInternalComments to true.</p>
</dd>
<dt><strong><a name="tofasta" class="item">toFasta</a></strong></dt>

<dd>
<p>Return the entry in Fasta format.</p>
</dd>
<dt><strong><a name="equal" class="item">equal</a></strong></dt>

<dd>
<p>Returns True if two entries are equal, False otherwise</p>
</dd>
</dl>
<p>The following methods are provided for your convenience. They are shortcuts for methods of the individual line objects.</p>
<dl>
<dt><strong><a name="id" class="item">ID</a></strong></dt>

<dd>
<p>Returns the primary ID of the entry.</p>
</dd>
<dt><strong><a name="ac" class="item">AC</a></strong></dt>

<dd>
<p>Returns the primary AC of the entry.</p>
</dd>
<dt><strong><a name="sq" class="item">SQ</a></strong></dt>

<dd>
<p>Returns the sequence of the entry.</p>
</dd>
<dt><strong><a name="ev" class="item">EV</a></strong></dt>

<dd>
<p>Returns the EV (evidence) object of an entry. SWISS-PROT internal method.</p>
</dd>
</dl>
<p>
</p>
<h2><a name="data_access_methods">Data access methods</a></h2>
<dl>
<dt><strong><a name="text" class="item">text</a></strong></dt>

<dd>
<p>Returns the current text of the entry. 
<strong>Quick and dirty!</strong> No update of the text is performed before.</p>
</dd>
<dt><strong><a name="database_code" class="item">database_code</a></strong></dt>

<dd>
<p>Is it a SWISS-PROT, TREMBL or TREMBLNEW entry?
database_code tries to find it out.
Return values are S for SWISS-PROT, 3 for TREMBL, Q for TREMBLNEW, ? for unknown.</p>
</dd>
<dt><strong><a name="isfragment" class="item">isFragment</a></strong></dt>

<dd>
<p>Returns true if the DE line indicates a fragment, or of the entry 
contains a NON_CONS or NON_TER feature.</p>
</dd>
<dt><strong><a name="iscurated" class="item">isCurated</a></strong></dt>

<dd>
<p>Returns 1 if the entry is a curated entry, 
0 otherwise.</p>
<p>SWISS-PROT internal use only.</p>
</dd>
</dl>

</body>

</html>
