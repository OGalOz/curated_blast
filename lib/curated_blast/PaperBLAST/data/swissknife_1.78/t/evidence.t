## Swissknife Test Harness Script for evidence tags
##

# * Test loading
BEGIN { 
  $| = 1; print "1..2\n"; 
  use vars qw($loaded); 
  $^W = 0;
}

END {print "not ok 1\n" unless $loaded;}


$loaded = 1;
print "ok 1\n";    # 1st test passes.

sub test ($$;$) {
    my($num, $true,$msg) = @_;
    print($true ? "ok $num\n" : "not ok $num $msg\n");
}

my $where = -d 't' ? "t/" : "";
my $testin = "${where}evidence.txl";
my $testout  = "${where}evidence.txl.out";
my $expectedout = "${where}evidence.txl.expected";

open (IN, $testin);
open (OUT, ">$testout");

use SWISS::Entry;

# Read an entire record at a time
$/ = "\/\/\n";

while (<IN>){
  # Read the entry
  $entry = SWISS::Entry->fromText($_, 1);
 
  #${$entry->DEs->list}[3]->deleteEvidenceTag('EC3'); #evidence tagging the DE elements currently not supported
  #${$entry->DEs->list}[2]->addEvidenceTag('EC3');
  foreach $ref ($entry->Refs->elements()) {
    $ref->addEvidenceTag('ECO:0000269|PubMed:11433298');
  }

  $ev1 = $entry->Stars->EV->addEvidence('ECO:0000269', 
					'PubMed:11433298', 
					'XXX', 
					'28-Aug-2014'); # p.s. EV->add/updateEvidence now only work with new style evidences!...
  $ev2 = $entry->Stars->EV->addEvidence('ECO:0000269', 
					'PubMed:12665801', 
					'XXX', 
					'28-Aug-2014');
  $ev3 = $entry->Stars->EV->addEvidence('ECO:0000312', 
					'EMBL:EAL60914.1', 
					'XXX', 
					'28-Aug-2014');
  
  $entry->DEs->setEvidenceTags($ev3);
  $entry->DEs->addEvidenceTag($ev2);
  if ($entry->DEs->hasEvidenceTag($ev2)){
	print OUT "Has evidence $ev2.\n";
  }	
  # test changed as no longer makes sense for revised CC module
  
  @CCs = $entry->CCs->elements();
  foreach $cc (@CCs) {
    if (!$cc->isa('SWISS::ListBase')) {
      $cc->addEvidenceTag($ev1);
    }
  }
  
  foreach $dr ($entry->DRs->elements) {
    $entry->DRs->addEvidenceTag($dr, 'ECO:0000269|PubMed:11433298');
    $entry->DRs->deleteEvidenceTag($dr, 'ECO:0000303');
  }
  
  # Check conditional delete
  $entry->DRs->add(['ZFIN', 'P123', 'Q1234', ' {ECO:0000312|EMBL:EAL60914.1}']);
  $entry->DRs->add(['ZFIN', 'P123', 'Q1234', ' {ECO:0000269|PubMed:12527781}']);

  foreach $ft ($entry->FTs->elements) {
    $entry->FTs->addEvidenceTag($ft, 'ECO:0000269|PubMed:11433298');
    $entry->FTs->deleteEvidenceTag($ft, 'ECO:0000305');
  }
  foreach $kw ($entry->KWs->elements) {
    $kw->addEvidenceTag($ev2);
    $kw->deleteEvidenceTag('EC2');
  }
  my $newkw = new SWISS::KW;
  $newkw->text('Key1');
  $entry->KWs->add($newkw);
  $newkw->setEvidenceTags($ev3);
  $newkw = new SWISS::KW;
  $newkw->text('Key2');
  $newkw->setEvidenceTags($ev3);
  $newkw->deleteEvidenceTag($ev3);  
  $entry->KWs->add($newkw);

  my $newTax = new SWISS::OX;
  $newTax->text(3332);
  $entry->OXs->NCBI_TaxID->add($newTax);

  foreach $os ($entry->OSs->elements) {
    $os->addEvidenceTag('ECO:0000269|PubMed:11433298');
    $os->deleteEvidenceTag('ECO:0000269|PubMed:11433298');
  }

  # Test if the Swiss-Prot style evidence (POTENTIAL etc) and the
  # evidence tags are properly separated. 
  foreach $ft ($entry->FTs->elements()) {
    print OUT join ",", @$ft, "\n";
  }
  print OUT $entry->toText;
}

close IN;
close OUT;
print "checking expected output...\n";

test 2, system('diff', $testout, $expectedout) == 0, "diff $testout $expectedout";





