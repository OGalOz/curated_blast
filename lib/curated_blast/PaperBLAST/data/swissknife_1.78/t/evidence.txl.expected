Has evidence ECO:0000269|PubMed:12665801.
TRANSIT,1,29,MITOCHONDRION,,, {ECO:0000303, ECO:0000269|PubMed:11433298},
CHAIN,30,430,ASPARTATE AMINOTRANSFERASE,,, {ECO:0000303, ECO:0000269|PubMed:11433298},
BINDING,279,279,PYRIDOXAL PHOSPHATE,BY SIMILARITY,, {ECO:0000269|PubMed:11433298},
VARIANT,3,3,A -> L,,/FTId=VAR_000001, {ECO:0000269|PubMed:11433298},
VARIANT,3,3,A -> LFSJFSHLJSDHFLSJLFHSJLLSJKDHFLSKJFHJ,,/FTId=VAR_000002, {ECO:0000269|PubMed:11433298},
ID   AATM_BOVIN     STANDARD;      PRT;   430 AA.
AC   P12344;
DT   01-OCT-1989 (Rel. 12, Created)
DT   01-NOV-1995 (Rel. 32, Last sequence update)
DT   15-JUL-1999 (Rel. 38, Last annotation update)
DE   ASPARTATE AMINOTRANSFERASE, MITOCHONDRIAL PRECURSOR (EC 2.6.1.1)
DE   (TRANSAMINASE A) (GLUTAMATE OXALOACETATE TRANSAMINASE-2)
DE   {ECO:0000312|EMBL:EAL60914.1, ECO:0000269|PubMed:12665801}.
GN   GOT2{EI1}.
OG   Plasmid pCP301 {ECO:0000313|EMBL:AAH50488.1,
OG   ECO:0000269|PubMed:11285225}, Plasmid pWR100, Plasmid pINV_F6_M1382,
OG   and Plasmid pSF5; Chloroplast.
OS   Bos taurus (Bovine){EI1}.
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;
OC   Mammalia; Eutheria; Cetartiodactyla; Ruminantia; Pecora; Bovoidea;
OC   Bovidae; Bovinae; Bos.
OX   NCBI_TaxID=126566, 55{EC2}, 3332;
RN   [1] {EI1, ECO:0000269|PubMed:11433298}
RP   SEQUENCE FROM N.A.
RC   TISSUE=HEART;
RA   Palmisano A.;
RL   Submitted (OCT-1993) to the EMBL/GenBank/DDBJ databases.
RN   [2] {EI1, ECO:0000269|PubMed:11433298}
RP   SEQUENCE OF 30-41.
RX   MEDLINE=79191877; PubMed=446759;
RA   Capasso S., Garzillo A.M., Marino G., Mazzarella L., Pucci P.,
RA   Sannia G.;
RT   "Mitochondrial bovine aspartate aminotransferase. Preliminary sequence
RT   and crystallographic data.";
RL   FEBS Lett. 101:351-354(1979).
CC   -!- CATALYTIC ACTIVITY: L-ASPARTATE + 2-OXOGLUTARATE = OXALOACETATE +
CC       L-GLUTAMATE{EC3}. {ECO:0000269|PubMed:11433298}.
CC   -!- COFACTOR:
CC       Note=PYRIDOXAL PHOSPHATE.{EC3};
CC   -!- SEQUENCE CAUTION:
CC       Sequence=CAI21743; Type=Erroneous gene model prediction;
CC       Sequence=CAI22254{EC1,EI1}; Type=Miscellaneous discrepancy; Positions=Several; Note=Translated as Gln;
CC   -!- RNA EDITING: Modified_positions=1, 4, 18, 33
CC       {ECO:0000269|PubMed:10707984, ECO:0000269|PubMed:11285225}, 34,
CC       72, 80, 86, 95, 121, 123, 154, 155, 156, 163, 169, 193, 196, 233,
CC       295, 346; Note=The initiator methionine is created by RNA editing.
CC       The nonsense codons at positions 72, 121, 169, 193 and 346 are
CC       modified to sense codons. {ECO:0000269|PubMed:10707984};
DR   EMBL; Z25466; CAA80960.1; -. {ECO:0000269|PubMed:11433298}
DR   ZFIN; P123; Q1234. {ECO:0000312|EMBL:EAL60914.1}
DR   ZFIN; P123; Q1234. {ECO:0000269|PubMed:12527781}
DR   PROSITE; PS00105; AA_TRANSFER_CLASS_1; 1. {ECO:0000269|PubMed:11433298}
KW   Transferase {ECO:0000269|PubMed:12665801};
KW   Aminotransferase {ECO:0000269|PubMed:12665801};
KW   Key1 {ECO:0000312|EMBL:EAL60914.1}; Key2.
FT   TRANSIT       1     29       MITOCHONDRION. {ECO:0000303,
FT                                ECO:0000269|PubMed:11433298}.
FT   CHAIN        30    430       ASPARTATE AMINOTRANSFERASE. {ECO:0000303,
FT                                ECO:0000269|PubMed:11433298}.
FT   BINDING     279    279       PYRIDOXAL PHOSPHATE (BY SIMILARITY).
FT                                {ECO:0000269|PubMed:11433298}.
FT   VARIANT       3      3       A -> L. {ECO:0000269|PubMed:11433298}.
FT                                /FTId=VAR_000001.
FT   VARIANT       3      3       A -> LFSJFSHLJSDHFLSJLFHSJLLSJKDHFLSKJFHJ
FT                                . {ECO:0000269|PubMed:11433298}.
FT                                /FTId=VAR_000002.
**
**   #################    INTERNAL SECTION    ##################
**EV EC2; Curator; MM; 06-SEP-1989.
**EV EC3; Curator; TT; 10-JUN-1999.
**EV ECO:0000269; PubMed:11433298; XXX; 28-Aug-2014.
**EV ECO:0000269; PubMed:12665801; XXX; 28-Aug-2014.
**EV ECO:0000312; EMBL:EAL60914.1; XXX; 28-Aug-2014.
**EV EI1; EMBL; CAA80960.1; 03-SEP-1989.
**OX 9913;
SQ   SEQUENCE   430 AA;  47513 MW;  16DDF475382035AA CRC64;
     MALLHSGRFL SGVAAAFHPG LAAAASARAS SWWAHVEMGP PDPILGVTEA FKRDTNSKKM
     NLGVGAYRDD NGKPYVLPSV RKAEAQIAAK NLDKEYLPIA GLAEFCKASA ELALGENNEV
     LKSGRYVTVQ TISGTGALRI GASFLQRFFK FSRDVFLPKP TWGNHTPIFR DAGMQLQSYR
     YYDPKTCGFD FTGAIEDISK IPAQSVILLH ACAHNPTGVD PRPEQWKEMA TVVKKNNLFA
     FFDMAYQGFA SGDGNKDAWA VRHFIEQGIN VCLCQSYAKN MGLYGERVGA FTVVCKDAEE
     AKRVESQLKI LIRPMYSNPP INGARIASTI LTSPDLRKQW LHEVKGMADR IISMRTQLVS
     NLKKEGSSHN WQHIIDQIGM FCYTGLKPEQ VERLTKEFSI YMTKDGRISV AGVTSGNVAY
     LAHAIHQVTK
//
